PubSub/src/BatchPublisher.php

```diff
@@ -141,16 +143,16 @@ public function publishDeferred(array $messages, array $options = [])
             $calls[$key][] = $message;
         }
 
-        if (!array_key_exists($this->topicName, self::$topics)) {
+        if (!array_key_exists($this->identifier, self::$topics)) {
             if (!$this->client) {
                 //@codeCoverageIgnoreStart
                 $this->client = new PubSubClient($this->getUnwrappedClientConfig());
                 //@codeCoverageIgnoreEnd
             }
-            self::$topics[$this->topicName] = $this->client->topic($this->topicName);
+            self::$topics[$this->identifier] = $this->client->topic($this->topicName);
         }
 
-        $topic = self::$topics[$this->topicName];
+        $topic = self::$topics[$this->identifier];
```

Comment:
I think in order to ensure the uniqueness, you need to change the `ID_TEMPLATE` as well.

Reply:
When the use case arises where the user needs something more unique than the default template thing, they would be supplying the identifier themselves and in that case the default one which uses the template won't be required.

Reply:
Yeah I like it better than us changing the default template. Better than our last conversation.